tests/test001.sh

```diff
@@ -1,101 +1,44 @@
-#! /bin/bash
+#!/bin/bash
 #
 # Minimal test established a peer link between two meshd instances over hwsim
 #
 
-[ $(uname) = "Linux" ] || { echo "This test only runs on Linux"; exit 1; }
+. `dirname $0`/include.sh
+
+[ $(uname) = "Linux" ] || err_exit "This test only runs on Linux"
 
 cleanup() {
     sudo killall meshd-nl80211
     rm -fr "${TMP1}" "${TMP2}"
     exit 0
 }
 
-get_hwsim_radios() {
-    local radios=""
-    for dev in /sys/devices/virtual/mac80211_hwsim/*; do
-        phy=$(basename $dev/ieee80211/*)
-        radios="$radios $phy"
-    done
-    echo $radios
-}
-
 trap cleanup SIGINT
 
-sudo modprobe -r mac80211-hwsim &> /dev/null
-sudo modprobe mac80211-hwsim radios=4 || { echo "Failed to load mac80211-hwsim module."; echo 1; }
-
-read -a radios <<<$(get_hwsim_radios)
-ifaces=("smesh0", "smesh1")
-
-rfkill unblock wifi
-sudo iw phy ${radios[0]} interface add ${ifaces[0]} type mesh
-sudo iw phy ${radios[1]} interface add ${ifaces[1]} type mesh
-
-sudo ip link set ${ifaces[0]} up
-sudo ip link set ${ifaces[1]} up
-
-MESHD=$(dirname $(realpath $0))/../build/linux/meshd-nl80211
-
-[ -x "${MESHD}" ] || { echo "${MESHD}" not found.; exit 1; }
-
-CONFIG=/tmp/authsae.cfg
-cat > ${CONFIG} <<EOF
-authsae:
-{
- sae:
-  {
-    debug = 480;
-    password = "thisisreallysecret";
-    group = [19, 26, 21, 25, 20];
-    blacklist = 5;
-    thresh = 5;
-    lifetime = 3600;
-  };
- meshd:
-  {
-    meshid = "byteme";
-    interface = "mesh0";
-    band = "11g";
-    channel = 1;
-    htmode = "none";
-    mcast-rate = 12;
-  };
-};
-EOF
-
-LOG0=/tmp/authsae-0.log
-LOG1=/tmp/authsae-1.log
-sudo rm -f ${LOG0} ${LOG1}
-sudo ${MESHD} -i ${ifaces[0]} -c ${CONFIG} -o ${LOG0} -B
-sudo ${MESHD} -i ${ifaces[1]} -c ${CONFIG} -o ${LOG1} -B
-wait
+nradios=2
+load_hwsim $nradios || err_exit "Failed to load mac80211-hwsim module."
+set_default_config
+start_meshd $(get_hwsim_radios) || exit 2
 
 # Wait for peer link establishment
 TRIES=50
-for i in $(seq $TRIES)
-do
-    grep established ${LOG0} &> /dev/null && break
-    echo -n .
-    sleep 1
-done
-
-for j in $(seq $TRIES)
-do
-    grep established ${LOG1} &> /dev/null && break
-    echo -n .
-    sleep 1
+for i in $(seq 0 $((nradios-1))); do
```

Comment:
Consider inverting these for loops so that we don't end up adding up to 50 seconds of sleep for each extra radio?

Reply:
Well, ok I see what you mean in the failure case, but I still left it :)